### DISCUSSION TITLE
Resetting event properties

### DISCUSSION THREAD

**Discussion by ekoranda** [imixs/imixs-workflow#725] [Q&A]
Hello Ralph, I'm wondering if it's possible to reset the event property items "mail.to_list_mapping" and "keymailreceiverfields". Both of these items contain old email addresses that I no longer want, however I can't figure out how to remove them. These emails are not included in my model at all and so I'm not sure why they are still mapped to my mailing list.

**Comment by rsoika** [comment]
Hi, where do you see these attributes? In the bpmn model file or in your process instances?

**Comment by ekoranda** [comment]
I was just calling #{workflowController.getEvents().get(0).getAllItems()} from one of the HTML files in my application (sub_main.xhtml) when I was trying to figure out why emails were being sent to the old email addresses.

**Comment by rsoika** [comment]
ok, I understand. In general if you change the field mapping in your model also those fields should be updated when you save the model. It's interesting that you say this does not happen. 
Which Imixs-BPMN Modeler version do you have currently installed?

![image](https://user-images.githubusercontent.com/1339124/85452811-8ab69000-b59b-11ea-9a84-677facec8d94.png)

We know that there is in some rare cases a strange behavior. If you just click/unclick the current selection this should solve the problem. 

![image](https://user-images.githubusercontent.com/1339124/85453236-f7318f00-b59b-11ea-95b7-54deb621f231.png)

But as I said, this should not happen. So maybe we can investigate into this if there is still a bug in the Imixs-BPMN modeller tool.

**Comment by ekoranda** [comment]
I currently have version 1.4.8.201906060621 of the IMIXS-BPMN modeler. I still have an issue when I tried clicking and unclicking the current selection. Even when I have no groups actors selected : 
<img width="1004" alt="Screen Shot 2020-06-23 at 3 23 01 PM" src="https://user-images.githubusercontent.com/47868453/85457249-6e97fc00-b565-11ea-88a4-194902d93ab9.png">, an email is still sent to the email addresses kept inside of "mail.to_list_mapping".

**Comment by rsoika** [comment]
hm... sounds like there is still a bug.

To fix the problem, just open the .bpmn file with a xml editor and search for the corresponding entry. Than remove the entry. 

If you can send me a reduced example of your bpmn model with the 'hidden' email entry we can file a [bug](https://github.com/imixs/imixs-bpmn/issues) and try to fix this issue.

**Comment by ekoranda** [comment]
Thanks Ralph going in with the xml editor and deleting the entry seemed to work. 
Here is another entry from my .bpmn file where the values of emily and Scott are old actors that I am no longer using but still seem to be 'hidden' entries. 

```
<imixs:item name="keymailreceiverfields" type="xs:string">
       <imixs:value><![CDATA[emily]]></imixs:value>
         <imixs:value><![CDATA[Scott]]></imixs:value>
          <imixs:value><![CDATA[group.proposerEmail]]></imixs:value>
</imixs:item>
```

**Comment by rsoika** [comment]
Ok, I will look into this.
How does your field-mapping for the workflow model look like? Can you make a screenshot so I can better understand the difference between the configured field-mappings and the Messaging-Receifer dialog page.

**Comment by ekoranda** [comment]
Here is the code for my event 10: 
```
<bpmn2:intermediateCatchEvent id="IntermediateCatchEvent_5" imixs:activityid="10" name="Submit">
      <bpmn2:extensionElements>
        <imixs:item name="txtactivityresult" type="CDATA">
          <imixs:value><![CDATA[<item name="action">worklist</item>

]]></imixs:value>
        </imixs:item>
        <imixs:item name="keyrestrictedvisibility" type="xs:string">
          <imixs:value><![CDATA[{imixs}]]></imixs:value>
          <imixs:value><![CDATA[{proposer1, proposer2}]]></imixs:value>
        </imixs:item>
        <imixs:item name="keyupdateacl" type="xs:boolean">
          <imixs:value>false</imixs:value>
        </imixs:item>
        <imixs:item name="keyownershipfields" type="xs:string">
          <imixs:value><![CDATA[{imixs}]]></imixs:value>
        </imixs:item>
        <imixs:item name="keyaddreadfields" type="xs:string">
          <imixs:value><![CDATA[{imixs}]]></imixs:value>
        </imixs:item>
        <imixs:item name="keyaddwritefields" type="xs:string">
          <imixs:value><![CDATA[{imixs}]]></imixs:value>
        </imixs:item>
        <imixs:item name="rtfresultlog" type="CDATA">
          <imixs:value><![CDATA[ticket submitted by <itemvalue>namcurrentEditor</itemvalue>]]></imixs:value>
        </imixs:item>
        <imixs:item name="txtmailsubject" type="xs:string">
          <imixs:value><![CDATA[to reviewer]]></imixs:value>
        </imixs:item>
        <imixs:item name="rtfmailbody" type="CDATA">
          <imixs:value><![CDATA[]]></imixs:value>
        </imixs:item>
        <imixs:item name="nammailreceiver" type="xs:string"/>
        <imixs:item name="$readaccess" type="xs:string">
          <imixs:value><![CDATA[group.proposer]]></imixs:value>
        </imixs:item>
        <imixs:item name="namaddreadaccess" type="xs:string"/>
        <imixs:item name="namaddwriteaccess" type="xs:string"/>
        <imixs:item name="keymailreceiverfields" type="xs:string">
          <imixs:value><![CDATA[group.reviewerEmail]]></imixs:value>
        </imixs:item>
      </bpmn2:extensionElements>
      <bpmn2:incoming>SequenceFlow_14</bpmn2:incoming>
      <bpmn2:outgoing>SequenceFlow_36</bpmn2:outgoing>
    </bpmn2:intermediateCatchEvent>
```

And here are some screenshots:
<img width="1004" alt="Screen Shot 2020-06-24 at 10 27 15 AM" src="https://user-images.githubusercontent.com/47868453/85585066-4eb81500-b605-11ea-9bd5-ac372d4166fb.png">
I declare group.propser in my application from an already existing ldap. 

<img width="1004" alt="Screen Shot 2020-06-24 at 10 27 52 AM" src="https://user-images.githubusercontent.com/47868453/85585140-5e375e00-b605-11ea-8e71-d5671cbc2191.png">
<img width="1004" alt="Screen Shot 2020-06-24 at 10 28 04 AM" src="https://user-images.githubusercontent.com/47868453/85585181-655e6c00-b605-11ea-9e94-097e2c8e0706.png">

The acl is not updated for this event.

**Comment by rsoika** [comment]
Ok I can reproduce this issue now. Thanks for providing the infos.

**Comment by rsoika** [comment]
I think I found the problem and I have already provided a fix.
If you click into the menu 'Help --> Check for Update' you will see a new version of Imixs-BPMN available:

![image](https://user-images.githubusercontent.com/1339124/85628484-8019fa80-b670-11ea-89e3-67a171bdafce.png)


If you change the 'Actor Properties' for a BPMN model, the existing Messaging Receivers will be now updates/removed accordingly. 

But the new Version can not fix an already 'corrupted' model file. So you need to verify your model withe the xml editor and remove the deprecated entries in this case. But for the future you should not run again into this trouble. 

**Background:** we already fixed the issue some time ago but only for the ACL settings and we forgot the messaging tab, which is also using actor-properties. https://github.com/imixs/imixs-bpmn/issues/71

**Comment by ekoranda** [comment]
Thank you that seems to be working!
